onnxruntime/core/providers/rocm/tunable/rocm_tuning_context.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+
+#include "core/framework/tuning_context.h"
+#include "core/providers/rocm/rocm_execution_provider_info.h"
+
+namespace onnxruntime {
+
+class ROCMExecutionProvider;
+
+namespace rocm {
+namespace tunable {
+
+class RocmTuningContext : public ITuningContext {
+ public:
+  explicit RocmTuningContext(ROCMExecutionProvider* ep, TunableOpInfo* info);
+
+  void EnableTunableOp() override;
+  void DisableTunableOp() override;
+  bool IsTunableOpEnabled() const override;
+
+  TuningResultsManager& GetTuningResultsManager() override;
+  const TuningResultsManager& GetTuningResultsManager() const override;
+
+ private:
+  TunableOpInfo* info_;  // non-owning handle
```

Comment:
> `TunableOpInfo* info_; // non-owning handle`

Consider using a `shared_ptr` (here and everywhere else) for `TunableOpInfo` objects.

Reply:
It is basically a POD (or aggregate) live in the EP, not a pointer to an object on the heap. So it does not make sense to be used with smart pointer... Or we might move the `TunableOpInfo` into this context, but it might make EP level access inconvenient.

Reply:
That makes sense. The current implementation is fine in that case.